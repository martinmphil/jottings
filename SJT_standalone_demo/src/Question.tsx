import React from "react";

interface Props {
  submissionError: boolean;
  questionNumber: number;
  scenarioText: string;
  optTextA: string;
  optTextB: string;
  optTextC: string;
  optTextD: string;
  submitHandling: () => void;
  selectBest: (x: number) => void;
  selectWorst: (x: number) => void;
  best: number;
  worst: number;
}

const Question: React.FC<Props> = props => {
  const isBestButtonDisabled = (optionNumber: number) => {
    // disable if this option is already selected as worst option
    if (optionNumber === props.worst) {
      return true;
    }
    // enable if no best option is selected
    else if (props.best === -1) {
      return false;
    }
    // enable if this option is selected thus allowing resetting on 2nd press
    else if (optionNumber === props.best) {
      return false;
    } else {
      return true;
    }
  };

  const isWorstButtonDisabled = (optionNumber: number) => {
    // disable if this option is already selected as worst option
    if (optionNumber === props.best) {
      return true;
    }
    // enable if no best option is selected
    else if (props.worst === -1) {
      return false;
    }
    // enable if this option is selected thus allowing resetting on 2nd press
    else if (optionNumber === props.worst) {
      return false;
    } else {
      return true;
    }
  };

  const selectedBestStyle = (optionNumber: number) => {
    // inline style overrides selected button class style
    if (optionNumber === props.best) {
      return { backgroundColor: "white" };
    }
  };

  const selectedWorstStyle = (optionNumber: number) => {
    // inline style overrides selected button class style
    if (optionNumber === props.worst) {
      return { backgroundColor: "white" };
    }
  };

  const handleBestClick = (x: number) => {
    props.selectBest(x);
  };

  const handleWorstClick = (x: number) => {
    props.selectWorst(x);
  };

  return (
    <main>
      <h1>Question {props.questionNumber}</h1>
      <p id="scenarioText">{props.scenarioText}</p>

      <hr />

      <div id="answerOptionA" className="answer-option">
        <button
          type="button"
          id="bestOptA"
          className={
            isBestButtonDisabled(0)
              ? "answer-button button-disabled"
              : "answer-button best-button"
          }
          style={selectedBestStyle(0)}
          aria-label="Option A is best"
          onClick={() => handleBestClick(0)}
          disabled={isBestButtonDisabled(0)}
        >
          Best
        </button>
        <span id="optTextA" className="option-text" aria-label="Option A">
          {props.optTextA}
        </span>
        <button
          type="button"
          id="worstOptA"
          className={
            isWorstButtonDisabled(0)
              ? "answer-button button-disabled"
              : "answer-button worst-button"
          }
          style={selectedWorstStyle(0)}
          aria-label="Option A is worst"
          onClick={() => handleWorstClick(0)}
          disabled={isWorstButtonDisabled(0)}
        >
          Worst
        </button>
      </div>

      <hr />

      <div id="answerOptionB" className="answer-option">
        <button
          type="button"
          id="bestOptB"
          className={
            isBestButtonDisabled(1)
              ? "answer-button button-disabled"
              : "answer-button best-button"
          }
          style={selectedBestStyle(1)}
          aria-label="Option B is best"
          onClick={() => handleBestClick(1)}
          disabled={isBestButtonDisabled(1)}
        >
          Best
        </button>
        <span id="optTextB" className="option-text" aria-label="Option B">
          {props.optTextB}
        </span>
        <button
          type="button"
          id="worstOptB"
          className={
            isWorstButtonDisabled(1)
              ? "answer-button button-disabled"
              : "answer-button worst-button"
          }
          style={selectedWorstStyle(1)}
          aria-label="Option B is worst"
          onClick={() => handleWorstClick(1)}
          disabled={isWorstButtonDisabled(1)}
        >
          Worst
        </button>
      </div>

      <hr />

      <div id="answerOptionC" className="answer-option">
        <button
          type="button"
          id="bestOptC"
          className={
            isBestButtonDisabled(2)
              ? "answer-button button-disabled"
              : "answer-button best-button"
          }
          style={selectedBestStyle(2)}
          aria-label="Option C is best"
          onClick={() => handleBestClick(2)}
          disabled={isBestButtonDisabled(2)}
        >
          Best
        </button>
        <span id="optTextC" className="option-text" aria-label="Option C">
          {props.optTextC}
        </span>
        <button
          type="button"
          id="worstOptC"
          className={
            isWorstButtonDisabled(2)
              ? "answer-button button-disabled"
              : "answer-button worst-button"
          }
          style={selectedWorstStyle(2)}
          aria-label="Option C is worst"
          onClick={() => handleWorstClick(2)}
          disabled={isWorstButtonDisabled(2)}
        >
          Worst
        </button>
      </div>

      <hr />

      <div id="answerOptionD" className="answer-option">
        <button
          type="button"
          id="bestOptD"
          className={
            isBestButtonDisabled(3)
              ? "answer-button button-disabled"
              : "answer-button best-button"
          }
          style={selectedBestStyle(3)}
          aria-label="Option D is best"
          onClick={() => handleBestClick(3)}
          disabled={isBestButtonDisabled(3)}
        >
          Best
        </button>
        <span id="optTextD" className="option-text" aria-label="Option D">
          {props.optTextD}
        </span>
        <button
          type="button"
          id="worstOptd"
          className={
            isWorstButtonDisabled(3)
              ? "answer-button button-disabled"
              : "answer-button worst-button"
          }
          style={selectedWorstStyle(3)}
          aria-label="Option D is worst"
          onClick={() => handleWorstClick(3)}
          disabled={isWorstButtonDisabled(3)}
        >
          Worst
        </button>
      </div>
      <hr />

      <button type="submit" onClick={props.submitHandling}>
        Submit
      </button>
      {props.submissionError && (
        <span className="error-warning">
          Sadly we experienced a submission error. Please refresh this page, or
          try again later.
        </span>
      )}
    </main>
  );
};

export default Question;
