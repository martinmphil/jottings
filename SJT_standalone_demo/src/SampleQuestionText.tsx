interface QuestionText {
  scenarioText: string;
  optTextA: string;
  optTextB: string;
  optTextC: string;
  optTextD: string;
}

const SampleQuestionText: QuestionText = {
  scenarioText:
    "You have recently taken a call from a customer praising the helpful service offered by a colleague in your team. Do you:",
  optTextA: "Tell your colleague about the feedback on a one-to-one basis.",
  optTextB: "Inform your team leader and let them give the praise personally.",
  optTextC:
    "Wait until the team meeting tomorrow and announce the feedback to everyone.",
  optTextD:
    "Email the whole company so that everyone hears about the great feedback."
};

export default SampleQuestionText;
