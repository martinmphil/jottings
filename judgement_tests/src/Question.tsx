import React from "react";

interface Props {
  scenarioText: string;
  optText1: string;
  optText2: string;
  optText3: string;
  optText4: string;
  submitHandling: () => void;
  bestOpt1: () => void;
  bestOpt2: () => void;
  bestOpt3: () => void;
  bestOpt4: () => void;
}

const Question: React.FC<Props> = props => {
  return (
    <section>
      <hr />

      <h1>Question one</h1>
      <p id="scenarioText">{props.scenarioText}</p>

      <hr />

      <div id="answerOption1" className="answer-option">
        <button
          type="button"
          id="bestOpt1"
          className="answer-button best-button"
          aria-label="Option 1 is best"
          onClick={props.bestOpt1}
        >
          Best
        </button>
        <span id="optText1" className="option-text" aria-label="Option 1">
          {props.optText1}
        </span>
        <button
          type="button"
          id="worstOpt1"
          className="answer-button worst-button"
          aria-label="Option 1 is worst"
        >
          Worst
        </button>
      </div>

      <hr />

      <div id="answerOption2" className="answer-option">
        <button
          type="button"
          id="bestOpt2"
          className="answer-button best-button"
          aria-label="Option 2 is best"
          onClick={props.bestOpt2}
        >
          Best
        </button>
        <span id="optText2" className="option-text" aria-label="Option 2">
          {props.optText2}
        </span>
        <button
          type="button"
          id="worstOpt2"
          className="answer-button worst-button"
          aria-label="Option 2 is worst"
        >
          Worst
        </button>
      </div>

      <hr />

      <div id="answerOption3" className="answer-option">
        <button
          type="button"
          id="bestOpt3"
          className="answer-button best-button"
          aria-label="Option 3 is best"
          onClick={props.bestOpt4}
        >
          Best
        </button>
        <span id="optText3" className="option-text" aria-label="Option 3">
          {props.optText3}
        </span>
        <button
          type="button"
          id="worstOpt3"
          className="answer-button worst-button"
          aria-label="Option 3 is worst"
        >
          Worst
        </button>
      </div>

      <hr />

      <div id="answerOption4" className="answer-option">
        <button
          type="button"
          id="bestOpt4"
          className="answer-button best-button"
          aria-label="Option 4 is best"
          onClick={props.bestOpt4}
        >
          Best
        </button>
        <span id="optText4" className="option-text" aria-label="Option 4">
          {props.optText4}
        </span>
        <button
          type="button"
          id="worstOpt4"
          className="answer-button worst-button"
          aria-label="Option 4 is worst"
        >
          Worst
        </button>
      </div>
      <hr />

      <button type="submit" onClick={props.submitHandling}>
        Submit
      </button>
    </section>
  );
};

export default Question;
