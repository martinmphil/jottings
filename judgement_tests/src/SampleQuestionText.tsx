interface QuestionText {
  scenarioText: string;
  optText1: string;
  optText2: string;
  optText3: string;
  optText4: string;
}

const SampleQuestionText: QuestionText = {
  scenarioText:
    "You have recently taken a call from a customer praising the helpful service offered by a colleague in your team. Do you:",
  optText1: "Tell your colleague about the feedback on a one-to-one basis.",
  optText2: "Inform your team leader and let them give the praise personally.",
  optText3:
    "Wait until the team meeting tomorrow and announce the feedback to everyone.",
  optText4:
    "Email the whole company so that everyone hears about the great feedback."
};

export default SampleQuestionText;
