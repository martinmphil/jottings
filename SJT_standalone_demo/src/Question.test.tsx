import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent, cleanup } from "@testing-library/react";

import Question from "./Question";
import SampleQuestionText from "./SampleQuestionText";

afterEach(cleanup);

test("candidate can see scenario text", () => {
  const { getByLabelText, container } = render(
    <Question
      submissionError={false}
      questionNumber={1}
      scenarioText={SampleQuestionText.scenarioText}
      optTextA={SampleQuestionText.optTextA}
      optTextB={SampleQuestionText.optTextB}
      optTextC={SampleQuestionText.optTextC}
      optTextD={SampleQuestionText.optTextD}
      submitHandling={jest.fn()}
      selectBest={jest.fn()}
      selectWorst={jest.fn()}
      best={-1}
      worst={-1}
    />
  );
  expect(container.querySelector("#scenarioText")).toHaveTextContent(
    SampleQuestionText.scenarioText
  );
  expect(container.querySelector("#optTextA")).toHaveTextContent(
    SampleQuestionText.optTextA
  );
  expect(container.querySelector("#optTextB")).toHaveTextContent(
    SampleQuestionText.optTextB
  );
  expect(container.querySelector("#optTextC")).toHaveTextContent(
    SampleQuestionText.optTextC
  );
  expect(container.querySelector("#optTextD")).toHaveTextContent(
    SampleQuestionText.optTextD
  );
  expect(getByLabelText("Option A is best")).toHaveTextContent("Best");
  expect(getByLabelText("Option D is worst")).toHaveTextContent("Worst");
});

test("button presses pass correct argument", () => {
  const selectBestMock = jest.fn();
  const selectWorstMock = jest.fn();
  const submitMock = jest.fn();
  const { getByLabelText, getByText } = render(
    <Question
      submissionError={false}
      questionNumber={1}
      scenarioText={""}
      optTextA={""}
      optTextB={""}
      optTextC={""}
      optTextD={""}
      submitHandling={submitMock}
      selectBest={selectBestMock}
      selectWorst={selectWorstMock}
      best={-1}
      worst={-1}
    />
  );

  fireEvent.click(getByLabelText("Option A is best"));
  expect(selectBestMock).toHaveBeenCalledTimes(1);
  expect(selectBestMock).toBeCalledWith(0);
  fireEvent.click(getByLabelText("Option B is best"));
  expect(selectBestMock).toHaveBeenCalledTimes(2);
  expect(selectBestMock).toBeCalledWith(1);
  fireEvent.click(getByLabelText("Option C is best"));
  expect(selectBestMock).toHaveBeenCalledTimes(3);
  expect(selectBestMock).toBeCalledWith(2);
  fireEvent.click(getByLabelText("Option D is best"));
  expect(selectBestMock).toHaveBeenCalledTimes(4);
  expect(selectBestMock).toBeCalledWith(3);

  fireEvent.click(getByLabelText("Option A is worst"));
  expect(selectWorstMock).toHaveBeenCalledTimes(1);
  expect(selectWorstMock).toBeCalledWith(0);
  fireEvent.click(getByLabelText("Option B is worst"));
  expect(selectWorstMock).toHaveBeenCalledTimes(2);
  expect(selectWorstMock).toBeCalledWith(1);
  fireEvent.click(getByLabelText("Option C is worst"));
  expect(selectWorstMock).toHaveBeenCalledTimes(3);
  expect(selectWorstMock).toBeCalledWith(2);
  fireEvent.click(getByLabelText("Option D is worst"));
  expect(selectWorstMock).toHaveBeenCalledTimes(4);
  expect(selectWorstMock).toBeCalledWith(3);

  fireEvent.click(getByText("Submit"));
  expect(submitMock).toHaveBeenCalledTimes(1);
});

test("best=(-1), worst=(-1) matches snapshot", () => {
  const selectBestMock = jest.fn();
  const selectWorstMock = jest.fn();
  const submitMock = jest.fn();
  const { asFragment, getByLabelText } = render(
    <Question
      submissionError={false}
      questionNumber={1}
      scenarioText={SampleQuestionText.scenarioText}
      optTextA={SampleQuestionText.optTextA}
      optTextB={SampleQuestionText.optTextB}
      optTextC={SampleQuestionText.optTextC}
      optTextD={SampleQuestionText.optTextD}
      submitHandling={submitMock}
      selectBest={selectBestMock}
      selectWorst={selectWorstMock}
      best={-1}
      worst={-1}
    />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(getByLabelText("Option A is best")).not.toHaveAttribute("disabled");
  expect(getByLabelText("Option B is best")).not.toHaveAttribute("disabled");
  expect(getByLabelText("Option C is best")).not.toHaveAttribute("disabled");
  expect(getByLabelText("Option D is best")).not.toHaveAttribute("disabled");
  expect(getByLabelText("Option A is worst")).not.toHaveAttribute("disabled");
  expect(getByLabelText("Option B is worst")).not.toHaveAttribute("disabled");
  expect(getByLabelText("Option C is worst")).not.toHaveAttribute("disabled");
  expect(getByLabelText("Option D is worst")).not.toHaveAttribute("disabled");
});

test("best=0, worst=1 matches snapshot", () => {
  const selectBestMock = jest.fn();
  const selectWorstMock = jest.fn();
  const submitMock = jest.fn();
  const { asFragment, getByLabelText } = render(
    <Question
      submissionError={false}
      questionNumber={1}
      scenarioText={SampleQuestionText.scenarioText}
      optTextA={SampleQuestionText.optTextA}
      optTextB={SampleQuestionText.optTextB}
      optTextC={SampleQuestionText.optTextC}
      optTextD={SampleQuestionText.optTextD}
      submitHandling={submitMock}
      selectBest={selectBestMock}
      selectWorst={selectWorstMock}
      best={0}
      worst={1}
    />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(getByLabelText("Option A is best")).not.toHaveAttribute("disabled");
  expect(getByLabelText("Option B is best")).toHaveAttribute("disabled");
  expect(getByLabelText("Option C is best")).toHaveAttribute("disabled");
  expect(getByLabelText("Option D is best")).toHaveAttribute("disabled");
  expect(getByLabelText("Option A is worst")).toHaveAttribute("disabled");
  expect(getByLabelText("Option B is worst")).not.toHaveAttribute("disabled");
  expect(getByLabelText("Option C is worst")).toHaveAttribute("disabled");
  expect(getByLabelText("Option D is worst")).toHaveAttribute("disabled");
});

test("best=2, worst=3 matches snapshot", () => {
  const selectBestMock = jest.fn();
  const selectWorstMock = jest.fn();
  const submitMock = jest.fn();
  const { asFragment, getByLabelText } = render(
    <Question
      submissionError={false}
      questionNumber={1}
      scenarioText={SampleQuestionText.scenarioText}
      optTextA={SampleQuestionText.optTextA}
      optTextB={SampleQuestionText.optTextB}
      optTextC={SampleQuestionText.optTextC}
      optTextD={SampleQuestionText.optTextD}
      submitHandling={submitMock}
      selectBest={selectBestMock}
      selectWorst={selectWorstMock}
      best={2}
      worst={3}
    />
  );
  expect(asFragment()).toMatchSnapshot();
  expect(getByLabelText("Option A is best")).toHaveAttribute("disabled");
  expect(getByLabelText("Option B is best")).toHaveAttribute("disabled");
  expect(getByLabelText("Option C is best")).not.toHaveAttribute("disabled");
  expect(getByLabelText("Option D is best")).toHaveAttribute("disabled");
  expect(getByLabelText("Option A is worst")).toHaveAttribute("disabled");
  expect(getByLabelText("Option B is worst")).toHaveAttribute("disabled");
  expect(getByLabelText("Option C is worst")).toHaveAttribute("disabled");
  expect(getByLabelText("Option D is worst")).not.toHaveAttribute("disabled");
});
