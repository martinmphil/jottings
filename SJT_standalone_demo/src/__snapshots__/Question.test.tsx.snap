// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`best=(-1), worst=(-1) matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <h1>
      Question 1
    </h1>
    <p
      id="scenarioText"
    >
      You have recently taken a call from a customer praising the helpful service offered by a colleague in your team. Do you:
    </p>
    <hr />
    <div
      class="answer-option"
      id="answerOptionA"
    >
      <button
        aria-label="Option A is best"
        class="answer-button best-button"
        id="bestOptA"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option A"
        class="option-text"
        id="optTextA"
      >
        Tell your colleague about the feedback on a one-to-one basis.
      </span>
      <button
        aria-label="Option A is worst"
        class="answer-button worst-button"
        id="worstOptA"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <div
      class="answer-option"
      id="answerOptionB"
    >
      <button
        aria-label="Option B is best"
        class="answer-button best-button"
        id="bestOptB"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option B"
        class="option-text"
        id="optTextB"
      >
        Inform your team leader and let them give the praise personally.
      </span>
      <button
        aria-label="Option B is worst"
        class="answer-button worst-button"
        id="worstOptB"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <div
      class="answer-option"
      id="answerOptionC"
    >
      <button
        aria-label="Option C is best"
        class="answer-button best-button"
        id="bestOptC"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option C"
        class="option-text"
        id="optTextC"
      >
        Wait until the team meeting tomorrow and announce the feedback to everyone.
      </span>
      <button
        aria-label="Option C is worst"
        class="answer-button worst-button"
        id="worstOptC"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <div
      class="answer-option"
      id="answerOptionD"
    >
      <button
        aria-label="Option D is best"
        class="answer-button best-button"
        id="bestOptD"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option D"
        class="option-text"
        id="optTextD"
      >
        Email the whole company so that everyone hears about the great feedback.
      </span>
      <button
        aria-label="Option D is worst"
        class="answer-button worst-button"
        id="worstOptd"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <button
      type="submit"
    >
      Submit
    </button>
  </main>
</DocumentFragment>
`;

exports[`best=0, worst=1 matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <h1>
      Question 1
    </h1>
    <p
      id="scenarioText"
    >
      You have recently taken a call from a customer praising the helpful service offered by a colleague in your team. Do you:
    </p>
    <hr />
    <div
      class="answer-option"
      id="answerOptionA"
    >
      <button
        aria-label="Option A is best"
        class="answer-button best-button"
        id="bestOptA"
        style="background-color: white;"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option A"
        class="option-text"
        id="optTextA"
      >
        Tell your colleague about the feedback on a one-to-one basis.
      </span>
      <button
        aria-label="Option A is worst"
        class="answer-button button-disabled"
        disabled=""
        id="worstOptA"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <div
      class="answer-option"
      id="answerOptionB"
    >
      <button
        aria-label="Option B is best"
        class="answer-button button-disabled"
        disabled=""
        id="bestOptB"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option B"
        class="option-text"
        id="optTextB"
      >
        Inform your team leader and let them give the praise personally.
      </span>
      <button
        aria-label="Option B is worst"
        class="answer-button worst-button"
        id="worstOptB"
        style="background-color: white;"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <div
      class="answer-option"
      id="answerOptionC"
    >
      <button
        aria-label="Option C is best"
        class="answer-button button-disabled"
        disabled=""
        id="bestOptC"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option C"
        class="option-text"
        id="optTextC"
      >
        Wait until the team meeting tomorrow and announce the feedback to everyone.
      </span>
      <button
        aria-label="Option C is worst"
        class="answer-button button-disabled"
        disabled=""
        id="worstOptC"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <div
      class="answer-option"
      id="answerOptionD"
    >
      <button
        aria-label="Option D is best"
        class="answer-button button-disabled"
        disabled=""
        id="bestOptD"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option D"
        class="option-text"
        id="optTextD"
      >
        Email the whole company so that everyone hears about the great feedback.
      </span>
      <button
        aria-label="Option D is worst"
        class="answer-button button-disabled"
        disabled=""
        id="worstOptd"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <button
      type="submit"
    >
      Submit
    </button>
  </main>
</DocumentFragment>
`;

exports[`best=2, worst=3 matches snapshot 1`] = `
<DocumentFragment>
  <main>
    <h1>
      Question 1
    </h1>
    <p
      id="scenarioText"
    >
      You have recently taken a call from a customer praising the helpful service offered by a colleague in your team. Do you:
    </p>
    <hr />
    <div
      class="answer-option"
      id="answerOptionA"
    >
      <button
        aria-label="Option A is best"
        class="answer-button button-disabled"
        disabled=""
        id="bestOptA"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option A"
        class="option-text"
        id="optTextA"
      >
        Tell your colleague about the feedback on a one-to-one basis.
      </span>
      <button
        aria-label="Option A is worst"
        class="answer-button button-disabled"
        disabled=""
        id="worstOptA"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <div
      class="answer-option"
      id="answerOptionB"
    >
      <button
        aria-label="Option B is best"
        class="answer-button button-disabled"
        disabled=""
        id="bestOptB"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option B"
        class="option-text"
        id="optTextB"
      >
        Inform your team leader and let them give the praise personally.
      </span>
      <button
        aria-label="Option B is worst"
        class="answer-button button-disabled"
        disabled=""
        id="worstOptB"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <div
      class="answer-option"
      id="answerOptionC"
    >
      <button
        aria-label="Option C is best"
        class="answer-button best-button"
        id="bestOptC"
        style="background-color: white;"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option C"
        class="option-text"
        id="optTextC"
      >
        Wait until the team meeting tomorrow and announce the feedback to everyone.
      </span>
      <button
        aria-label="Option C is worst"
        class="answer-button button-disabled"
        disabled=""
        id="worstOptC"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <div
      class="answer-option"
      id="answerOptionD"
    >
      <button
        aria-label="Option D is best"
        class="answer-button button-disabled"
        disabled=""
        id="bestOptD"
        type="button"
      >
        Best
      </button>
      <span
        aria-label="Option D"
        class="option-text"
        id="optTextD"
      >
        Email the whole company so that everyone hears about the great feedback.
      </span>
      <button
        aria-label="Option D is worst"
        class="answer-button worst-button"
        id="worstOptd"
        style="background-color: white;"
        type="button"
      >
        Worst
      </button>
    </div>
    <hr />
    <button
      type="submit"
    >
      Submit
    </button>
  </main>
</DocumentFragment>
`;
